### PR TITLE
fix(settings): Move client-side email validation to fxa-shared

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We need to have the same email validation rules for both settings pages

## This pull request

- Moves our client-side email validation to fxa-shared so that it can be used in content-server and fxa-settings
- Adds email validation to the add secondary email page in new settings

## Issue that this pull request solves

Closes: # (issue number)

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by vbudhram** [comment]
Unrelated test failures in fxa-shared

```
+ yarn build
../fxa-shared/nestjs/customs/customs.service.ts:11:24 - error TS7016: Could not find a declaration file for module 'superagent'. '/home/circleci/project/node_modules/superagent/lib/node/index.js' implicitly has an 'any' type.
  Try `npm install @types/superagent` if it exists or add a new declaration (.d.ts) file containing `declare module 'superagent';`

11 import superagent from 'superagent';
                          ~~~~~~~~~~~~

../fxa-shared/nestjs/customs/customs.service.ts:51:12 - error TS7006: Parameter 'res' implicitly has an 'any' type.

51       .ok((res) => res.status < 600);
```

**Comment by dannycoates** [comment]
> Unrelated test failures in fxa-shared

@vbudhram would you mind pushing a fix to #7125 and merge it?

File: packages/fxa-content-server/app/scripts/lib/validate.js [hunk 1]
```diff
@@ -58,20 +44,7 @@ var Validate = {
...
-    // the regex, thus validating unicode addresses. However, doing
-    // that would break Firefox versions 45 and lower, because of
-    // https://bugzilla.mozilla.org/show_bug.cgi?id=1243594. So, in
-    // order to support unicode email addresses in the future, we'll
-    // have to pass the punycoded address in all our dealings with
-    // the browser and the non-punycoded address in our dealings with
-    // the auth server. :-/
-
-    return emailRegex.test(email);
+    return isEmailValid(email);
```
Comment (chenba) [inline packages/fxa-content-server/app/scripts/lib/validate.js:47]: Looks kinda odd. 😄   Can it just be `isEmailValid,`?

**Review by chenba — APPROVED** [review]
r+ with comment

**Pull request merged**